vmm_core/src/partition_unit/vp_set.rs

```diff
@@ -426,31 +432,68 @@ impl<T: Processor, U> DebugVp for BoundVp<'_, T, U> {
         Ok(())
     }
 
-    fn get_vp_state(&mut self, vtl: Vtl) -> anyhow::Result<Box<virt::x86::VpState>> {
+    #[cfg(guest_arch = "x86_64")]
+    fn get_vp_state(&mut self, vtl: Vtl) -> anyhow::Result<Box<DebuggerVpState>> {
         let mut access = self.vp.access_state(vtl);
         let regs = access.registers()?;
         let msrs = access.virtual_msrs()?;
-        Ok(Box::new(virt::x86::VpState {
-            gp: [
-                regs.rax, regs.rcx, regs.rdx, regs.rbx, regs.rsp, regs.rbp, regs.rsi, regs.rdi,
-                regs.r8, regs.r9, regs.r10, regs.r11, regs.r12, regs.r13, regs.r14, regs.r15,
-            ],
-            rip: regs.rip,
-            rflags: regs.rflags,
-            cr0: regs.cr0,
-            cr2: regs.cr2,
-            cr3: regs.cr3,
-            cr4: regs.cr4,
-            cr8: regs.cr8,
-            efer: regs.efer,
-            kernel_gs_base: msrs.kernel_gs_base,
-            es: regs.es,
-            cs: regs.cs,
-            ss: regs.ss,
-            ds: regs.ds,
-            fs: regs.fs,
-            gs: regs.gs,
-        }))
+        Ok(Box::new(DebuggerVpState::X86_64(
+            vmm_core_defs::debug_rpc::X86VpState {
+                gp: [
+                    regs.rax, regs.rcx, regs.rdx, regs.rbx, regs.rsp, regs.rbp, regs.rsi, regs.rdi,
+                    regs.r8, regs.r9, regs.r10, regs.r11, regs.r12, regs.r13, regs.r14, regs.r15,
+                ],
+                rip: regs.rip,
+                rflags: regs.rflags,
+                cr0: regs.cr0,
+                cr2: regs.cr2,
+                cr3: regs.cr3,
+                cr4: regs.cr4,
+                cr8: regs.cr8,
+                efer: regs.efer,
+                kernel_gs_base: msrs.kernel_gs_base,
+                es: regs.es,
+                cs: regs.cs,
+                ss: regs.ss,
+                ds: regs.ds,
+                fs: regs.fs,
+                gs: regs.gs,
+            },
+        )))
+    }
+
+    #[cfg(guest_arch = "aarch64")]
+    fn set_vp_state(&mut self, _vtl: Vtl, state: &DebuggerVpState) -> anyhow::Result<()> {
+        let DebuggerVpState::Aarch64(_state) = state else {
+            anyhow::bail!("wrong architecture")
+        };
+        anyhow::bail!("todo");
```

Comment:
is this hard to do, or just annoying to do? would be nice to just get it out of the way now if its just the latter

Reply:
I kept reading, and see that there's a solid amount of wiring here. Seems reasonable to punt for now.